tests/state/test_state_api.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from remoulade.api.main import app
+from remoulade.state import State, StateNamesEnum
+
+
+@pytest.fixture
+def api_client(state_middleware):
+    with app.test_client() as client:
+        yield client
+
+
+class TestMessageStateAPI:
+    """ Class Responsible to do the test of the API """
+
+    def test_no_messages(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/messages/states")
+        assert res.status_code == 200
+        assert len(res.json["result"]) == 0
+
+    def test_invalid_url(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/invalid_url/")
+        assert res.status_code == 404
+
+    def test_pending_message(self, stub_broker, state_middleware, api_client):
+        state_middleware.backend.set_state(State("id", StateNamesEnum.Pending, [], {}), ttl=1000)
+        res = api_client.get("/messages/states")
```

Comment:
Can you add like 3 or 4 states with different status and test for the full json ?